wipp-backend-data/src/main/java/gov/nist/itl/ssd/wipp/backend/data/pyramidannotation/timeslices/PyramidAnnotationTimeSliceController.java

```diff
@@ -150,8 +150,8 @@ public PyramidAnnotation uploadAnnotationPositions(
 		}
 
 		List<PyramidAnnotationTimeSlice> allTimeSlices = pyramidAnnotation.getTimeSlices();
-
-		if(containsTimeSlice(allTimeSlices, timeSliceId)) {
+		
+		if(pyramidAnnotationTimeSliceRepository.findOne(pyramidAnnotationId, timeSliceId) != null) {	
 			for (int i = 0; i < allTimeSlices.size(); i++) {
 				if(allTimeSlices.get(i).getSliceNumber() == timeSliceId) {
```

Comment:
You don't need to look for the timeSlice again here, you should get it from `pyramidAnnotationTimeSliceRepository.findOne`